src/shogun/mathematics/linalg/internal/modules/SpecialPurpose.h

```diff
@@ -67,15 +69,15 @@ void rectified_linear(Matrix A, Matrix result)
 	implementation::special_purpose::rectified_linear<backend, Matrix>::compute(A, result);
 }
 
-/** Performs the operation C(i,j) = C(i,j) * (A(i,j)!=0) for all i and j*/ 
+/** Performs the operation C(i,j) = C(i,j) * (A(i,j)!=0) for all i and j*/
 template <Backend backend=linalg_traits<Core>::backend,class Matrix>
 void multiply_by_rectified_linear_derivative(Matrix A, Matrix C)
 {
 	implementation::special_purpose::multiply_by_rectified_linear_derivative<backend, Matrix>::compute(A, C);
 }
 
-/** Applies the softmax function inplace to a matrix. The softmax function is 
- * defined as \f$ f(A[i,j]) = \frac{exp(A[i,j])}{\sum_i exp(A[i,j])} \f$ 
+/** Applies the softmax function inplace to a matrix. The softmax function is
+ * defined as \f$ f(A[i,j]) = \frac{exp(A[i,j])}{\sum_i exp(A[i,j])} \f$
  */
 template <Backend backend=linalg_traits<Core>::backend,class Matrix>
 void softmax(Matrix A)
```

Comment:
BTW, @karlnapf  in theory, it seems FITC could work for the multiclass logistic GP. Not sure whether it is robust or not. Maybe, I take a look at this later.